extension/test/unit/payment-handler.spec.js

```diff
@@ -0,0 +1,131 @@
+    }
+  }
+  /* eslint-enable max-len */
+
+  beforeEach(() => {
+    scope = nock(`${config.adyen.apiBaseUrl}`)
+  })
+
+  afterEach(() => {
+    nock.cleanAll()
+  })
+
+  it('when payment contains "submitPaymentDetailsRequest" with "makePaymentResponse", ' +
+    'then it should call "submitPaymentDetailsRequest"', async () => {
+    scope.post('/payments/details')
+      .reply(200, submitPaymentDetailsChallengeResponse)
+
+    const ctpPaymentClone = _.cloneDeep(ctpPayment)
+    ctpPaymentClone.custom.fields.makePaymentResponse = JSON.stringify(makePaymentRedirectResponse)
+    ctpPaymentClone.custom.fields.submitAdditionalPaymentDetailsRequest = JSON.stringify(submitPaymentDetailsRequest)
+
+    const response = await handlePayment(ctpPaymentClone)
+
+    expect(response.data.actions).to.have.lengthOf.above(0)
+  })
+
+  it('when payment does not contain "makePaymentResponse", ' +
+    'then it should not call "submitPaymentDetailsRequest"', async () => {
+    scope.post('/payments/details')
+      .reply(200, submitPaymentDetailsChallengeResponse)
+
+    const ctpPaymentClone = _.cloneDeep(ctpPayment)
+    ctpPaymentClone.custom.fields.submitAdditionalPaymentDetailsRequest = JSON.stringify(submitPaymentDetailsRequest)
+
+    const response = await handlePayment(ctpPaymentClone)
+
+    expect(response.data.actions).to.have.lengthOf(0)
+  })
+
+  it('when "submitPaymentDetailsRequest" contains same request, ' +
+    'then it should NOT call Adyen endpoint again if response exists', async () => {
```

Comment:
I would rather use specific test name in here rather than saying Adyen endpoint -> /payment/details/ endpoint
also, it looks like a wrong naming because the handler is the payment-handler is a facade/manager for all handler functions, so I would focus on which methods are called during payment-handler,  for example `then it should NOT call submitAdditionalPaymentDetailsHandler` wdyt?

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/271/commits/be080c246088f6cf37a36cf613e6d746d8eb2f45